libraries/math/src/processor.rs

```diff
@@ -111,3 +111,120 @@ pub fn process_instruction(
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::instruction::MathInstruction;
+    use borsh::BorshSerialize;
+    use solana_program::account_info::AccountInfo;
+
+    #[test]
+    fn test_u64_multiply() {
+        assert_eq!(4, u64_multiply(2, 2));
+        assert_eq!(12, u64_multiply(4, 3));
+    }
+
+    #[test]
+    fn test_u64_divide() {
+        assert_eq!(1, u64_divide(2, 2));
+        assert_eq!(2, u64_divide(2, 1));
+    }
+
+    #[test]
+    fn test_f32_multiply() {
+        assert_eq!(4.0, f32_multiply(2.0, 2.0));
+        assert_eq!(12.0, f32_multiply(4.0, 3.0));
+    }
+
+    #[test]
+    fn test_f32_divide() {
+        assert_eq!(1.0, f32_divide(2.0, 2.0));
+        assert_eq!(2.0, f32_divide(2.0, 1.0));
+    }
```

Comment:
nit: how about writing these tests with the math on the left-hand side, e.g.
```
assert_eq!(2 *2, u64_multiply(2, 2));
```
or
```
assert_eq!(2.0 / 2.0, f32_divide(2.0, 2.0));
```

Reply:
assert_eq!(2.0 / 2.0, f32_divide(2.0, 2.0)); gives a clippy error with default settings

```
error: equal expressions as operands to `/`
   --> libraries/math/src/processor.rs:130:20
    |
130 |         assert_eq!(2 / 2, u64_divide(2, 2));
    |                    ^^^^^
    |
    = note: `#[deny(clippy::eq_op)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#eq_op
```

I'll revert this.

Reply:
No worries, it wasn't very important.  That's pretty cool though, the `eq_op` lint checks for things like `assert_eq!(x, x)` which means that the compiler can figure out that you're doing `assert_eq!(2 / 2, 2 / 2)`!